srcs/fps.c

```diff
@@ -0,0 +1,54 @@
+#include "cub.h"
+
+int	display_fps(t_cub *cub)
+{
+	char				*time_str;
+	char				*prev_str;
+	long double			frame_time;
+
+	time_str = NULL;
+	prev_str = NULL;
+	cub->old_time = cub->time;
+	cub->time = get_time_micros();
+	frame_time = ((long double)(cub->time - cub->old_time)) / 1000000;
+	cub->prev_fps = cub->curr_fps;
+	cub->curr_fps = 1.0 / frame_time;
+	if (fabs(cub->curr_fps - cub->prev_fps) > 1)
+	{
+		time_str = set_fps_string(time_str, cub, 0);
+		prev_str = set_fps_string(prev_str, cub, 1);
+		mlx_string_put(cub->mlx, cub->window, 25, 25, 0x00000000, prev_str);
+		mlx_string_put(cub->mlx, cub->window, 25, 25, 0x00FFFFFF, time_str);
+		free(time_str);
+		free(prev_str);
+	}
+	return (0);
+}
+
+char	*set_fps_string(char *str1, t_cub *cub, int mode)
```

Comment:
why not just pass cub->curr_fps or cub->prev_fps (depending on the situation) instead of passing the whole cub and a mode? then you just say str = ft_itoa(whatever fps you passed as an argument of the function)
P.S: let me know if what I'm saying here is not clear, I feel like I'm not explaining super well

Reply:
I see what you mean. Hasn't been written very well and need to look into this more deeply...

Reply:
Ok! Maybe let's leave that for another branch then